internal/abci/abci.go

```diff
@@ -164,6 +172,29 @@ func NewAbciApp(ctx context.Context, cfg *AbciConfig, snapshotter SnapshotModule
 	if height == -1 {
 		height = 0 // negative means first start (no state table yet), but need non-negative for below logic
 	}
+
+	if bytes.Equal(appHash, []byte{0x42}) {
+		// This is an interesting corner case where the node crashes after the commit of block execution
+		// state but before the commit of the app hash. The chain state will have the app hash as "42".
+		// CometBFT seems to persist the FinalizeBlock responses, so during restart it calls "Commit"
+		// directly without calling FinalizeBlock and assumes that the application has the correct state
+		// persisted, whereas the application has the app hash as "42". This will cause the current block
+		// to be committed with an invalid app hash (sentry node blocksyncing will fail at this point).
+		// To fix this, we will store the last commit info in a file and use that as the application's
+		// app hash whenever we find the chain state to be inconsistent signified by the app hash as "42".
+		lc, err := loadLastCommitInfo(app.lastCommitInfoFileName)
+		if err != nil {
+			return nil, err
```

Comment:
I think we should decorate this error as it might be some cryptic complaint from the json decoder about an invalid utf8 character.  Provide context and even resolution (drop pg DB and rebuild state from existing block data)